pyfaidx/__init__.py

```diff
@@ -303,14 +303,25 @@ def __init__(self, filename, default_seq=None, key_function=lambda x: x.split()[
             else:
                 self._fasta_opener = bgzf.open
                 self._bgzf = True
-        elif filenameLower.endswith('.gz') or filenameLower.endswith('.bz2'):
-            raise RuntimeError("Compressed FASTA is only supported in BGZF "
-                               "format, via the BioPython bgzf module.")
+        elif filenameLower.endswith('.bz2') or filenameLower.endswith('.zip'):
+            raise RuntimeError(
+                "Compressed FASTA is only supported in BGZF format. Use "
+                "bgzip to compresss your FASTA.")
         else:
             self._fasta_opener = open
             self._bgzf = False
 
-        self.file = self._fasta_opener(filename, 'r+b' if mutable else 'rb')
+        try:
+            self.file = self._fasta_opener(filename, 'r+b' if mutable else 'rb')
+        except ValueError as e:
+            if str(e).find('BGZF') > -1:
+                raise RuntimeError(
+                    "Compressed FASTA is only supported in BGZF format. Use "
+                    "the samtools bgzip utility (instead of gzip) to "
+                    "compresss your FASTA.")
+            else:
+                raise
```

Comment:
I'm okay with your proposal. Testing the exception message *is* a bit pedantic, and it wouldn't be my top priority, but I wouldn't complain if you did it :)